src/puppetlabs/puppetdb/mq.clj

```diff
@@ -264,6 +266,42 @@
     (vec (repeatedly n #(commit-or-rollback s
                           (convert-jms-message (.receive consumer)))))))
 
+(def ^:private local-broker-mb
+  "org.apache.activemq:type=Broker,brokerName=localhost")
+
+(defn-validated local-queue-size
+  "Returns the number of pending messages in the named local queue.
+  Throws {:type ::queue-not-found :name name} when the queue doesn't exist."
+  [name :- s/Str]
+  (try
+    (jmx/read
+     (format "%s,destinationType=Queue,destinationName=%s" local-broker-mb name)
+     :QueueSize)
+    (catch javax.management.InstanceNotFoundException ex
+      (throw+ {:type ::queue-not-found :name name}))))
+
+(defn-validated transfer-local-messages!
+  "Transfers all of the messages currently available in the named
+  source queue to the destination and returns the number transferred.
+  Each message successfully transferred will be committed."
+  [connection :- connection-schema
+   source :- s/Str
+   destination :- s/Str]
+  (jmx/invoke-signature
+   (format "%s,destinationType=Queue,destinationName=%s" local-broker-mb source)
+   :moveMatchingMessagesTo
+   ["java.lang.String" "java.lang.String"]
+   "JMSMessageID is NOT NULL"
+   destination))
+
+(defn-validated remove-local-queue!
+  "Deletes the named local queue.  Throws
+  {:type ::queue-not-found :name name} when the queue doesn't exist."
+  [name :- s/Str]
+  ;; When the queue doesn't exist, removeQueue provides no indication.
+  (jmx/invoke-signature local-broker-mb
+                        :removeQueue ["java.lang.String"] name))
+
```

Comment:
Do we want any unit tests for these right now?  Related: https://tickets.puppetlabs.com/browse/PDB-1634